doc/user-manual/language/type-based-termination-checking.rst

```diff
@@ -0,0 +1,317 @@
+..
+  ::
+  module language.type-based-termination-checking where
+
+      open import Agda.Builtin.Nat
+      open import Agda.Builtin.List
+
+.. _type-based-termination-checking:
+
+*******************************
+Type-Based Termination Checking
+*******************************
+
+.. note::
+   Type-based termination checking is based on a master's thesis described in :ref:`[1]<type-based-termination-checking-references>`.
+
+Sometimes the default termination checker in Agda may seem too weak. Let's explore this in the context of defining a finitely-branching tree:
```

Comment:
I would use a more descriptive term than just "default", e.g. "syntax-based" or "structural". Also add a link to the page where the syntactic termination check is described!